pkg/apis/core/validation/validation.go

```diff
@@ -1533,6 +1533,34 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 	return allErrs
 }
 
+// ValidatePVSecretReference check whether provided SecretReference object is valid in terms of secret name and namespace.
```

Comment:
is it a good idea to have boolean args to enable/disable name and namespace ? as some of the secretRefs are localObjectReference?

Reply:
if you plan to use this function for other pv sources, yes I think it's a good idea. Similar idea: https://github.com/kubernetes/kubernetes/pull/103024#discussion_r692620224

Reply:
accommodating  all PV sources and making sure all combinations work and adding missing parts on all these sources is bit wide effort. i would like to start with CSI volume source though and progress one by one by filling the requirement.

Reply:
with above thought in place, could you please review this PR @roycaihw .. In short, main intention is to come up with a base function and validate CSI Volume Source atm. Other PV sources I would like to keep as out of scope in this PR's view.